File: basic/Eratosthenes.java
public class Eratosthenes {
    public static boolean[] sieve(int n) {
        boolean[] isPrime = new boolean[n + 1];

        for (int i = 2; i <= n; i++) {
            isPrime[i] = true;
        }

        for (int i = 2; i * i < n; i++) {
            if (isPrime[i]) {
                for (int j = i * i; j <= n; j += i) {
                    isPrime[j] = false;
                }
            }
        }
        return isPrime;
    }
    
    // 소수 출력을 위한 메인 메소드
    public static void main(String[] args) {
        int n = 100; // 1부터 100까지의 소수를 찾음
        boolean[] primes = sieve(n);
        
        System.out.println("1부터 " + n + "까지의 소수:");
        for (int i = 2; i <= n; i++) {
            if (primes[i]) {
                System.out.print(i + " ");
            }
        }
        System.out.println();
    }
}

File: basic/prime/Eratosthenes.cpp
#include <iostream>
#include <string>
#include <vector>
#include <cmath>

using namespace std;

std::vector<bool> sieve(int n) {
    std::vector<bool> isPrime(n + 1, true);

    isPrime[0] = false; // 0은 소수가 아님
    isPrime[1] = false; // 1은 소수가 아님

    for (int i = 2; i * i <= n; i++) {
        if (isPrime[i]) {
            for (int j = i * i; j <= n; j += i) {
                isPrime[j] = false;
            }
        }
    }

    return isPrime;
}

// 테스트 케이스 실행
void runTests() {
    vector<int> testCases = {100};

    for (size_t i = 0; i < testCases.size(); ++i) {
        int n = testCases[i];

        cout << "Test Case " << i + 1 << ": n=" << n << endl;
        vector<bool> primes = sieve(n);
        for (int i = 2; i <= n; i++) {
            if (primes[i]) {
                cout << i << " ";
            }
        }
        cout << endl;
    }
}

int main() {
    runTests();
    return 0;
}